ansible/eda-server
# Handle HealthyQueueNotFoundError for new processes.

- Handle HealthyQueueNotFoundError when a new process is going to be created (restart, start, autostart) 
- Update the status message when an activation is pending because there are no healthy nodes.
- Fixes a regresion where an activation running in a unhealthy node is scheduled automatically in a new node instead of set "workers-offline" status
- Fixes a bug where we don't check the health of the queue if only one queue (default queue for k8s and single-node deployments) was set. 

Test flow:
1. create an activation until reach running state. 
2. kill the activation workers of the node where the activation is running (from now node A), the activation goes into workers-offline state
3. (optional) relaunch workers to test recoverability
4. restart activation, it should be running in node B
5. kill workers of node B, activation goes into workers-offline. 
6. Create a new activation, it should go into pending state with proper status message
7. Restart the first activation in workers offline state, it should go into pending state with proper status message.
8. (optional) relaunch some activation worker to test that both activations runs. 


Replaces https://github.com/ansible/eda-server/pull/892
Fixes: https://issues.redhat.com/browse/AAP-23378 and https://issues.redhat.com/browse/AAP-22907
